src/key_value_store/server/kvstore_server.cpp

```diff
@@ -26,17 +26,29 @@ Status KeyValueStoreImpl::put(ServerContext *context, const PutRequest *request,
   }
 }
 
-Status
-KeyValueStoreImpl::get(ServerContext *context,
-                       ServerReaderWriter<GetReply, GetRequest> *stream) {
+Status KeyValueStoreImpl::get(
+    ServerContext *context, ServerReaderWriter<GetReply, GetRequest> *stream) {
   GetRequest request;
   while (stream->Read(&request)) {
-    // Get the value vector and store it in values
-    std::vector<std::string> values(store_.Get(request.key()));
-    for (const std::string &s : values) {
-      GetReply response;
-      response.set_value(s);
-      stream->Write(response);
+    // check if requestiong previous or stream of data
+    if (context->client_metadata().find("request-type")->second == "previous") {
+      // Get the value vector and store it in values
+      std::vector<std::string> values(store_.Get(request.key()));
+      for (const std::string &s : values) {
+        GetReply response;
+        response.set_value(s);
+        stream->Write(response);
+      }
+    } else {  // request-type == stream
+      std::function<bool(std::string)> sub_function = [stream](std::string m) {
```

Comment:
Choose a more descriptive variable name for "m", perhaps caw_string (this is going to be a caw that is serialized to a string)?

Reply:
good call